source/beamline/SGM/energy/SGMGratingTranslationStepControl.h

```diff
@@ -0,0 +1,77 @@
+#ifndef SGMGRATINGTRANSLATIONSTEPCONTROL_H
+#define SGMGRATINGTRANSLATIONSTEPCONTROL_H
+
+#include "beamline/AMPseudoMotorControl.h"
+/*!
+  * Pseudo-motor class which acts as a wrapper for the grating translation PV,
+  * which does not update its moving status if asked to move within tolerance.
+  */
+class SGMGratingTranslationStepControl : public AMPseudoMotorControl
```

Comment:
Is it necessary to have AMPseudoMotorControl as an ancestor for this class? It seems like this class gets most of its functionality from asking the gratingStepTranslationPV_ control for information. What additional functionality does this class provide?